app/src/main/java/one/credify/sdk/sample/requester/MarketRequester.kt

```diff
@@ -0,0 +1,112 @@
+package one.credify.sdk.sample.requester
+
+import android.content.Context
+import android.util.Log
+import com.android.volley.toolbox.JsonObjectRequest
+import com.android.volley.toolbox.StringRequest
+import com.android.volley.toolbox.Volley
+import com.google.gson.Gson
+import one.credify.sdk.sample.model.MarketInfo
+import one.credify.sdk.sample.response.AccessTokenResponse
+import org.json.JSONObject
+
+class MarketRequester {
+    fun getAccessToken(
+        context: Context,
+        url: String,
+        apiKey: String,
+        onRequest: () -> Unit,
+        onResult: (accessToken: String?) -> Unit
+    ) {
+        onRequest()
+
+        Log.d("MarketRequester", "Url: $url")
+
+        // Instantiate the RequestQueue.
+        val queue = Volley.newRequestQueue(context)
+
+        // Request a string response from the provided URL.
+        val stringRequest = object : StringRequest(
+            Method.POST,
+            url,
+            { response ->
+                // Display the first 500 characters of the response string.
+                val gson = Gson()
+                val accessToken = gson.fromJson(response, AccessTokenResponse::class.java)
+
+                onResult(accessToken.data?.accessToken)
+            },
+            {
+                // Error
+                Log.d("MarketRequester", "$it")
+                onResult(null)
+            }
+        ) {
+            override fun getHeaders(): MutableMap<String, String> {
+                return HashMap<String, String>().apply {
+                    put("X-API-KEY", apiKey)
+                }
+            }
+        }
+
+        queue.add(stringRequest)
+    }
+
+    fun pushClaimToken(
+        context: Context,
+        marketInfo: MarketInfo,
+        localId: String,
+        credifyId: String,
+        onRequest: () -> Unit,
+        onResult: (isSuccess: Boolean) -> Unit
+    ) {
+        getAccessToken(
```

Comment:
Why is this access token needed?

Reply:
@ngo275 To exchange `apiKey` to `accessToken` and using that `accessToken` to push claim tokens

Reply:
`push claim tokens` of demo organization server doesn't require Credify's access token.

Reply:
@ngo275 I updated it.